mesa/mesaPROTON_OC.py

```diff
@@ -607,10 +607,43 @@ def setup_schools(self):
         """
         Generates n-schools based on the number of initial agents
         """
-        for level in m.education_levels.keys():
-            for i_school in range(int(m.education_levels[level][3])):
-                new_school = School(self, level, list())
-                m.schools.append(new_school)
+        for level in self.education_levels.keys():
+            for i_school in range(int(self.education_levels[level][3])):
+                new_school = School(self, level)
+                self.schools.append(new_school)
+
+    def init_students(self):
+        """
+        Adds to schools the agents that meet the defined parameters of age and level of education and then
+        creates connections between agents within the school.
+        """
+        for level in self.education_levels:
+            row = self.education_levels[level]
+            start_age = row[0]
+            end_age = row[1]
+            pool = [x for x in self.schedule.agents if
+                    x.age() >= start_age and x.age() <= end_age and x.education_level == level - 1 and x.max_education_level >= level]
+            for agent in pool:
+                agent.enroll_to_school(level)
+        for school in self.schools:
+            conn = self.decide_conn_number(school.my_students, 15)
+            for student in school.my_students:
+                total_pool = school.my_students.difference({student})
```

Comment:
Just use `remove`.

Reply:
The "remove" method removes agents in-place, we cannot modify a set while it is iterated, for this reason I used the "difference" method.